lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlobFileSystem.cs

```diff
@@ -0,0 +1,255 @@
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Auth;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Org.Apache.REEF.Tang.Annotations;
+
+namespace Org.Apache.REEF.IO.FileSystem.AzureBlob
+{
+    /// <summary>
+    /// An IFileSystem implementation for Azure Blob storage.
+    /// A few calls are quite costly due to the inability to distinguish what type
+    /// of blob we are trying to access, so GetBlobReferenceFromServer is invoked.
+    /// There is currently no GetBlobReference call that does not populate
+    /// the fields of an ICloudBlob.
+    /// </summary>
+    public sealed class AzureBlobFileSystem : IFileSystem
+    {
+        private readonly ICloudBlobClient _client;
+
+        public AzureBlobFileSystem(Uri baseUri)
+        {
+            _client = new AzureCloudBlobClient(baseUri);
+        }
+
+        public AzureBlobFileSystem(Uri baseUri, StorageCredentials storageCredentials)
+        {
+            _client = new AzureCloudBlobClient(baseUri, storageCredentials);
+        }
+
+        public AzureBlobFileSystem(StorageUri storageUri, StorageCredentials storageCredentials)
+        {
+            _client = new AzureCloudBlobClient(storageUri, storageCredentials);
+        }
+
+        public AzureBlobFileSystem(string connectionString)
+        {
+            _client = new AzureCloudBlobClient(connectionString);
+        }
+
+        [Inject]
+        private AzureBlobFileSystem(ICloudBlobClient client)
+        {
+            _client = client;
+        } 
+
+        /// <summary>
+        /// Not supported for Azure Blobs, will throw <see cref="NotSupportedException"/>.
+        /// </summary>
+        public Stream Open(Uri fileUri)
+        {
+            throw new NotSupportedException("Open is not supported for AzureBlobFileSystem.");
+        }
+
+        /// <summary>
+        /// Not supported for Azure Blobs, will throw <see cref="NotSupportedException"/>.
+        /// </summary>
+        public Stream Create(Uri fileUri)
+        {
+            throw new NotSupportedException("Open is not supported for AzureBlobFileSystem.");
+        }
+
+        /// <summary>
+        /// <see cref="IFileSystem.Delete"/>
+        /// </summary>
+        public void Delete(Uri fileUri)
+        {
+            var uriSplit = fileUri.AbsolutePath.Split(new[] { "/" }, StringSplitOptions.RemoveEmptyEntries);
+            if (uriSplit.Length <= 1)
```

Comment:
+1 for changing exception type as well. It is an artifact of your implementation of splitting uri that it is argumentoutofrange. In semantics it is either invalidoperationexception or argumentexception?